### PR TITLE
feat(TDP-11227): update CI for release builds

### PR DESCRIPTION
**Pull request by anicolas**
CI updates to be able to release the TCOMPv0 service from the `maintenance/tdp` branch

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -45,14 +64,69 @@ pipeline {
 
   parameters {
     choice(name: 'Action',
-      choices: ['STANDARD', 'DEPLOY'],
-      description: 'Kind of running : \nSTANDARD (default), normal building \nDEPLOY : build & deploy TCOMPv0 service Docker image (TDP cloud) & zip artefact (TDP on-prem)')
+      choices: ['STANDARD', 'PUBLISH', 'RELEASE'],
+      description: 'Type of build : \nSTANDARD (default): compile & run tests \nPUBLISH : build & deploy TCOMPv0 service Docker image (TDP cloud) & zip artefact (TDP on-prem)\nRELEASE: build, deploy & release TCOMPv0 service Docker image (TDP cloud) & zip artefact (TDP on-prem)'
```
Comment (eguillossou) [inline ci/Jenkinsfile:68]: Can you adapt all description to publish as well? 
`PUBLISH : build & publish TCOMPv0 service Docker image (TDP cloud)  & zip artefact (TDP on-prem) to artifactory` maybe?

File: ci/Jenkinsfile [hunk 2]
```diff
@@ -63,70 +137,102 @@ pipeline {
...
+                --threads 1C \
+                --settings "${MAVEN_SETTINGS}"
+               '''
+          }
         }
       }
     }
 
-    stage('Deploy service Docker image') {
+    stage('Build & deploy service Docker image') {
```
Comment (eguillossou) [inline ci/Jenkinsfile:190]: Stage name need to be updated to `stage('Build & publish service Docker image') {`

File: ci/Jenkinsfile [hunk 3]
```diff
@@ -63,70 +137,102 @@ pipeline {
...
+                    '${dockerAndGitTag}' \
+                    '${MAVEN_SETTINGS}'
+                """
+            }
           }
         }
       }
     }
 
     stage('Deploy service zip artefact') {
```
Comment (eguillossou) [inline ci/Jenkinsfile:217]: idem

File: ci/Jenkinsfile (hunk 1 above)
Comment (anicolas) [reply ci/Jenkinsfile]: Nice catch! Done 😉

File: ci/Jenkinsfile (hunk 2 above)
Comment (anicolas) [reply ci/Jenkinsfile]: Done 😉

File: ci/Jenkinsfile (hunk 3 above)
Comment (anicolas) [reply ci/Jenkinsfile]: Also done 😉

**Review by eguillossou — APPROVED** [review]

**Pull request merged**